src/NodeTypeResolver/Reflection/BetterReflection/SourceLocatorProvider/DynamicSourceLocatorProvider.php

```diff
@@ -76,6 +79,21 @@ public function provide(): SourceLocator
 
         $this->aggregateSourceLocator = new AggregateSourceLocator($sourceLocators);
 
+        $reflector = new DefaultReflector($this->aggregateSourceLocator);
+        foreach ($sourceLocators as $sourceLocator) {
+            // trigger collect "classes" on get class on locate identifier
+            // which call Reflector->reflectClass()
+            try {
+                $sourceLocator->locateIdentifiersByType($reflector, new class() extends IdentifierType {
+                    public function isClass(): bool
+                    {
+                        return true;
+                    }
+                });
+            } catch (CouldNotReadFileException) {
+            }
+        }
```

Comment:
@TomasVotruba this is what I mean, previously, on very first run, the `classes` seems overlapped by which one file that scanned first, then on 2nd run, the class child will be detected.

This ensure `Reflector->reflectClass()` called on all `ClassLike` so it exists in the `classes` property in `MemoizingReflectionProvider`

I will try add back `getChildrenOfClassReflection()` method to `FamilyRelationsAnalyzer`, add fixture test, then let @nuryagdym manually try the patch.